src/SceneGraphDebugCommandController.ts

```diff
@@ -0,0 +1,305 @@
+import { util } from './util';
+
+// eslint-disable-next-line
+const Telnet = require('telnet-client');
+
+export class SceneGraphDebugCommandController {
+    constructor(public host: string) {
+    }
+
+    private connection: typeof Telnet;
+
+    private shellPrompt = '>';
+    private echoLines = 0;
+    private timeout = 5000;
+    private port = 8080;
+
+    public async connect() {
+        this.removeConnection();
+
+        try {
+            // Make a new telnet connections object
+            let connection = new Telnet();
+
+            connection.on('close', this.removeConnection);
```

Comment:
This won't work. `this` gets lost when you pass the function by reference like this. You need to do:
```typescript
connection.on('close', this.removeConnection.bind(this))
```

or

```typescript
connection.on('close', ()=>{
    this.removeConnection();
});
```

Reply:
Will update.